ouroboros-network/test/Test/Ouroboros/Network/Testnet.hs

```diff
@@ -677,6 +694,113 @@ prop_diffusion_target_active_local_above defaultBearerInfo diffScript =
                    <*> demotionOpportunities
                    <*> demotionOpportunitiesIgnoredTooLong)
 
+
+-- | A variant of ouroboros-network-framework
+-- 'Test.Ouroboros.Network.Server2.prop_connection_manager_valid_transitions'
+-- but for running on Diffusion. This means it has to have in consideration the
+-- the logs for all nodes running will all appear in the trace and the test
+-- property should only be valid while a given node is up and running.
+--
+-- We do not need separate above and below variants of this property since it
+-- is not possible to exceed the target.
```

Comment:
What do you mean by _above and below variants_?

Reply:
the same test but for above and below conditions.